### DISCUSSION TITLE
How to publish several slides to the same GitHub Pages site

### DISCUSSION THREAD

**Discussion by SoumayaMauthoorMOJ** [marp-team/marp#524] [Show and tell]
Hello :wave:

I like to use marp to summarise investigations into new technologies because it means all the code and documentation resides in a single location. The investigations are saved to the same [repo](https://github.com/moj-analytical-services/dmet-cfe) which means I can't publish a GitHub Page per investigation.

Instead, I use a  [`marp-to-pages`](https://github.com/moj-analytical-services/dmet-cfe/blob/main/.github/workflows/gh-deploy.yml) GitHub workflow which loops through the folder investigations, copies any images in the respective `images` folder, converts the slides to HTML and publishes the slides under `https://moj-analytical-services.github.io/dmet-cfe/<investigation_name>`

**Comment by yhatt** [comment]
Marp team has no particular opinion about the structure for managing and publishing multiple Marp slides, but it would become easy to publish multiple pages to GitHub Pages by matching the structure of Markdown and related files into the Web page.

For example, if you are managing Markdown documents with a following strcuture:

```
.
└── src/
    ├── slide-1/
    │   ├── index.md
    │   └── assets/
    │       ├── a.png
    │       ├── b.png
    │       └── ...
    ├── slide-2/
    │   ├── index.md
    │   └── assets/
    │       └── ...
    └── slide-3/
        └── ...
```

You can prepare the artifact for publishing to GitHub pages with the following simple build steps powered by Marp CLI:

```bash
cp -r ./src ./dist
marp ./dist
```

Marp CLI will find out Markdown files from `./dist`, and save converted HTML files into next to each HTML.

```
.
└── dist/
    ├── slide-1/
    │   ├── index.md
    │   ├── index.html
    │   └── assets/
    │       └── ...
    ├── slide-2/
    │   ├── index.md
    │   ├── index.html
    │   └── assets/
    │       └── ...
    └── slide-3/
        └── ...
```

Then, you can publish all slides to GitHub Pages by uploading `dist` directory as [the page artifact for GitHub Pages](https://github.com/actions/upload-pages-artifact). Each slides will be available as `https://<username>.github.io/<repository>/slide-1/`, `https://<username>.github.io/<repository>/slide-2/`, and so on.

Also refer to [the help for GitHub Pages for publishing pages with the custom GitHub Actions](https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site).

**Reply by SoumayaMauthoorMOJ** [reply]
thanks @yhatt I'll try that out :-)

**Reply by SoumayaMauthoorMOJ** [reply]
Hey @yhatt that worked really thanks for the tip! I've used it [here](https://github.com/moj-analytical-services/data-and-analytics-engineering-tech-radar/blob/main/.github/workflows/gh_deploy.yml)

I decided to keep the slides as README.me to ensure that GitHub renders them properly. I then added a step to rename the README.md to index.md:

```
      - name: Rename README.md to index.md
        run: find ./build -type f -name "README.md" -execdir mv {} index.md \;
```
